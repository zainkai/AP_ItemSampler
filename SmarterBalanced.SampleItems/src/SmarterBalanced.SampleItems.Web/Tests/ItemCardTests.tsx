import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ItemCard, ItemCardViewModel } from '../Scripts/ItemCard';
import { GradeLevels } from '../scripts/GradeLevels';

describe('ItemCard', () => {
    test('should display the correct information', () => {
        const viewModel: ItemCardViewModel = {
            bankKey: 1,
            itemKey: 1,
            title: "Deciding What to Eat 101",
            grade: GradeLevels.Grade3,
            gradeLabel: "Grade 3",
            subjectCode: "LNCH",
            subjectLabel: "LNCH/eating",
            claimCode: "LNCH1",
            claimLabel: "Reading",
            targetHash: "1234",
            targetShortName: "Key Details",
            interactionTypeCode: "MC",
            interactionTypeLabel: "Multiple Choice"
        }

        const itemCard = renderer.create(
            <ItemCard {...viewModel} key={Math.random()} />
        );
        const tree = itemCard.toJSON();
        expect(tree).toMatchSnapshot();
    });
});