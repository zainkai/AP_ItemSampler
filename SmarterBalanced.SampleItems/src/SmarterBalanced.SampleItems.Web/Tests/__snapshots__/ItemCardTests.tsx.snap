// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard should display the correct information 1`] = `
<div
  className="card card-block lnch"
  onClick={[Function]}
  onKeyUp={[Function]}
  tabIndex={0}
>
  <div
    className="card-contents"
  >
    <h4
      className="card-title"
      onClick={[Function]}
      onKeyUp={[Function]}
    >
      Deciding What to Eat 101
    </h4>
    <p
      className="card-text subject"
    >
      <span
        className="card-text-label"
      >
        Subject:
      </span>
      <span
        className="card-text-value"
      >
         
        LNCH/eating
      </span>
    </p>
    <p
      className="card-text grade"
    >
      <span
        className="card-text-label"
      >
        Grade:
      </span>
      <span
        className="card-text-value"
      >
         
        Grade 3
      </span>
    </p>
    <p
      className="card-text claim"
    >
      <span
        className="card-text-label"
      >
        Claim:
      </span>
      <span
        className="card-text-value"
      >
         
        Reading
      </span>
    </p>
    <p
      className="card-text target"
    >
      <span
        className="card-text-label"
      >
        Target:
      </span>
      <span
        className="card-text-value"
      >
         
        Key Details
      </span>
    </p>
    <p
      className="card-text interaction-type"
    >
      <span
        className="card-text-label"
      >
        Item Type:
      </span>
      <span
        className="card-text-value"
      >
         
        Multiple Choice
      </span>
    </p>
    <p
      className="card-text item-id"
    >
      <span
        className="card-text-label"
      >
        Item Id:
      </span>
      <span
        className="card-text-value"
      >
         
        1
      </span>
    </p>
  </div>
</div>
`;
